test/src/test/java/org/corfudb/runtime/collections/PersistedCorfuTableTest.java

```diff
@@ -0,0 +1,1006 @@
+        resetTests();
+        try (final PersistedCorfuTable<String, String> table = setupTable()) {
+            intended.forEach(value -> table.insert(value, value));
+            intended.forEach(k -> Assertions.assertEquals(k, table.get(k)));
+        }
+    }
+
+    /**
+     * Verify commit address of transactions for disk-backed tables.
+     */
+    @Property(tries = NUM_OF_TRIES)
+    void verifyCommitAddressMultiTable(@ForAll @Size(SAMPLE_SIZE) Set<String> intended) {
+        resetTests();
+        try (final PersistedCorfuTable<String, String> table1 = setupTable();
+             final PersistedCorfuTable<String, String> table2 = setupTable(alternateTableName)) {
+            table1.insert(defaultNewMapEntry, defaultNewMapEntry);
+
+            assertThat(executeTx(() -> {
+                table1.get(nonExistingKey);
+                table2.get(nonExistingKey);
+            })).isZero();
+
+            intended.forEach(value -> table2.insert(value, value));
+            assertThat(executeTx(() -> {
+                table1.get(nonExistingKey);
+                table2.get(nonExistingKey);
+            })).isEqualTo(intended.size());
+        }
+    }
+
+    /**
+     * Verify commit address of interleaving transactions on disk-backed tables.
+     */
+    // TODO(ZacH):
+    @Property(tries = NUM_OF_TRIES)
+    void verifyCommitAddressInterleavingTxn(@ForAll @Size(SAMPLE_SIZE) Set<String> intended) throws Exception {
+        resetTests();
+        try (final PersistedCorfuTable<String, String> table = setupTable()) {
+            CountDownLatch latch1 = new CountDownLatch(1);
+            CountDownLatch latch2 = new CountDownLatch(1);
+
+            Thread t1 = new Thread(() -> {
+                table.insert(defaultNewMapEntry, defaultNewMapEntry);
+                assertThat(executeTx(() -> table.get(nonExistingKey))).isEqualTo(0L);
+                assertThat(executeTx(() -> {
+                    try {
+                        table.get(nonExistingKey);
+                        latch2.countDown();
+                        latch1.await();
+                        table.get(nonExistingKey);
+                    } catch (InterruptedException ignored) {
+                        // Ignored
+                    }
+                })).isEqualTo(intended.size());
+            });
+
+            Thread t2 = new Thread(() -> {
+                try {
+                    latch2.await();
+                    intended.forEach(value -> table.insert(value, value));
+                    assertThat(executeTx(() -> table.get(nonExistingKey))).isEqualTo(intended.size());
+                    latch1.countDown();
+                } catch (InterruptedException ex) {
+                    // Ignored
+                }
+            });
+
+            t1.start();
+            t2.start();
+            t1.join();
+            t2.join();
+            intended.forEach(value -> assertThat(table.get(value)).isEqualTo(value));
+        }
+    }
```

Comment:
Some of these tests are really targeting the read committed functionality, but there is value in also testing snapshot isolation. Could we do both for such tests?

Reply:
Done.